In [1]:
import os
import pandas as pd

# structure

In [2]:
ma1 = pd.read_csv('hyperparam-tuning-results/structure-mdeberta-macro_l1.csv', index_col=0)
ma2 = pd.read_csv('hyperparam-tuning-results/structure-mdeberta-macro_l2.csv', index_col=0)
mi1 = pd.read_csv('hyperparam-tuning-results/structure-mdeberta-micro_l1.csv', index_col=0)
mi2 = pd.read_csv('hyperparam-tuning-results/structure-mdeberta-micro_l2.csv', index_col=0)

In [3]:
len(ma1), len(ma2), len(mi1), len(mi2)

(5, 5, 5, 5)

In [4]:
ma1[['test_overall_accuracy', 'test_overall_f1']].mean()

test_overall_accuracy    0.921668
test_overall_f1          0.456324
dtype: float64

In [5]:
ma2[['test_overall_accuracy', 'test_overall_f1']].mean()

test_overall_accuracy    0.856152
test_overall_f1          0.287610
dtype: float64

In [6]:
mi1[['test_overall_accuracy', 'test_overall_f1']].mean()

test_overall_accuracy    0.664502
test_overall_f1          0.205709
dtype: float64

In [7]:
mi2[['test_overall_accuracy', 'test_overall_f1']].mean()

test_overall_accuracy    0.632854
test_overall_f1          0.212358
dtype: float64

## adapter

In [8]:
ama1 = pd.read_csv('hyperparam-tuning-results/structure-mdeberta-adapter-macro_l1.csv', index_col=0)
ama2 = pd.read_csv('hyperparam-tuning-results/structure-mdeberta-adapter-macro_l2.csv', index_col=0)
ami1 = pd.read_csv('hyperparam-tuning-results/structure-mdeberta-adapter-micro_l1.csv', index_col=0)
ami2 = pd.read_csv('hyperparam-tuning-results/structure-mdeberta-adapter-micro_l2.csv', index_col=0)

In [9]:
len(ama1), len(ama2), len(ami1), len(ami2)

(5, 5, 5, 5)

In [10]:
ama1[['test_overall_accuracy', 'test_overall_f1']].mean()

test_overall_accuracy    0.947732
test_overall_f1          0.676351
dtype: float64

In [11]:
ama2[['test_overall_accuracy', 'test_overall_f1']].mean()

test_overall_accuracy    0.919337
test_overall_f1          0.515307
dtype: float64

In [12]:
ami1[['test_overall_accuracy', 'test_overall_f1']].mean()

test_overall_accuracy    0.763819
test_overall_f1          0.489982
dtype: float64

In [13]:
ami2[['test_overall_accuracy', 'test_overall_f1']].mean()

test_overall_accuracy    0.729821
test_overall_f1          0.455188
dtype: float64

### select representative fold

In [14]:
f1s = pd.DataFrame()

In [15]:
f1s['ma1'] = ama1[['test_overall_f1']]
f1s['ma2'] = ama2[['test_overall_f1']]
f1s['mi1'] = ami1[['test_overall_f1']]
f1s['mi2'] = ami2[['test_overall_f1']]

In [16]:
f1s

ma1       ma2       mi1       mi2
0  0.693642  0.512648  0.512753  0.479349
1  0.654545  0.546112  0.487609  0.428571
2  0.673854  0.509213  0.446782  0.438853
3  0.668493  0.493409  0.515406  0.465940
4  0.691218  0.515152  0.487360  0.463226

In [17]:
f1s.mean()

ma1    0.676351
ma2    0.515307
mi1    0.489982
mi2    0.455188
dtype: float64

In [18]:
abs(f1s['ma1'] - f1s['ma1'].mean()) # 2

0    0.017291
1    0.021805
2    0.002496
3    0.007857
4    0.014868
Name: ma1, dtype: float64

In [19]:
abs(f1s['ma2'] - f1s['ma2'].mean()) # 4

0    0.002659
1    0.030806
2    0.006094
3    0.021898
4    0.000155
Name: ma2, dtype: float64

In [20]:
abs(f1s['mi1'] - f1s['mi1'].mean()) # 1 (4 is close)

0    0.022771
1    0.002373
2    0.043200
3    0.025424
4    0.002622
Name: mi1, dtype: float64

In [21]:
abs(f1s['mi2'] - f1s['mi2'].mean()) # 4

0    0.024161
1    0.026616
2    0.016335
3    0.010752
4    0.008038
Name: mi2, dtype: float64

--> fold = 8 (row 4) is most respresentative across the dimensions

# quality

In [22]:
textfunktion = pd.read_csv('hyperparam-tuning-results/quality-mdeberta-textfunktion.csv', index_col=0)
inhaltliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-mdeberta-inhaltliche_ausgestaltung.csv', index_col=0)
textstruktur = pd.read_csv('hyperparam-tuning-results/quality-mdeberta-textstruktur.csv', index_col=0)
sprachliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-mdeberta-sprachliche_ausgestaltung.csv', index_col=0)
gesamteindruck = pd.read_csv('hyperparam-tuning-results/quality-mdeberta-gesamteindruck.csv', index_col=0)

In [23]:
len(textfunktion), len(inhaltliche_ausgestaltung), len(textstruktur), len(sprachliche_ausgestaltung), len(gesamteindruck)

(5, 5, 5, 5, 5)

In [24]:
textfunktion[['test_qwk']].mean()

test_qwk    0.530306
dtype: float64

In [25]:
inhaltliche_ausgestaltung[['test_qwk']].mean()

test_qwk    0.294719
dtype: float64

In [26]:
textstruktur[['test_qwk']].mean()

test_qwk    0.512792
dtype: float64

In [27]:
sprachliche_ausgestaltung[['test_qwk']].mean()

test_qwk    0.492092
dtype: float64

In [28]:
gesamteindruck[['test_qwk']].mean()

test_qwk    0.615869
dtype: float64

In [29]:
# standard deviation
textfunktion[['test_qwk']].std(), inhaltliche_ausgestaltung[['test_qwk']].std(), textstruktur[['test_qwk']].std(), sprachliche_ausgestaltung[['test_qwk']].std(), gesamteindruck[['test_qwk']].std(), 

(test_qwk    0.068692
 dtype: float64,
 test_qwk    0.108741
 dtype: float64,
 test_qwk    0.043634
 dtype: float64,
 test_qwk    0.059088
 dtype: float64,
 test_qwk    0.039733
 dtype: float64)

## adapter

In [30]:
atextfunktion = pd.read_csv('hyperparam-tuning-results/quality-mdeberta-adapter-textfunktion.csv', index_col=0)
ainhaltliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-mdeberta-adapter-inhaltliche_ausgestaltung.csv', index_col=0)
atextstruktur = pd.read_csv('hyperparam-tuning-results/quality-mdeberta-adapter-textstruktur.csv', index_col=0)
asprachliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-mdeberta-adapter-sprachliche_ausgestaltung.csv', index_col=0)
agesamteindruck = pd.read_csv('hyperparam-tuning-results/quality-mdeberta-adapter-gesamteindruck.csv', index_col=0)

In [31]:
len(atextfunktion), len(ainhaltliche_ausgestaltung), len(atextstruktur), len(asprachliche_ausgestaltung), len(agesamteindruck)

(5, 5, 5, 5, 5)

In [32]:
atextfunktion[['test_qwk']].mean()

test_qwk    0.564146
dtype: float64

In [33]:
ainhaltliche_ausgestaltung[['test_qwk']].mean()

test_qwk    0.430512
dtype: float64

In [34]:
atextstruktur[['test_qwk']].mean()

test_qwk    0.575175
dtype: float64

In [35]:
asprachliche_ausgestaltung[['test_qwk']].mean()

test_qwk    0.578677
dtype: float64

In [36]:
agesamteindruck[['test_qwk']].mean()

test_qwk    0.647826
dtype: float64

In [37]:
# standard deviation
atextfunktion[['test_qwk']].std(), ainhaltliche_ausgestaltung[['test_qwk']].std(), atextstruktur[['test_qwk']].std(), asprachliche_ausgestaltung[['test_qwk']].std(), agesamteindruck[['test_qwk']].std(), 

(test_qwk    0.018057
 dtype: float64,
 test_qwk    0.097673
 dtype: float64,
 test_qwk    0.038412
 dtype: float64,
 test_qwk    0.077051
 dtype: float64,
 test_qwk    0.054022
 dtype: float64)

## fusion

In [38]:
ftextfunktion = pd.read_csv('hyperparam-tuning-results/quality-fusion-textfunktion.csv', index_col=0)
finhaltliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-inhaltliche_ausgestaltung.csv', index_col=0)
ftextstruktur = pd.read_csv('hyperparam-tuning-results/quality-fusion-textstruktur.csv', index_col=0)
fsprachliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-sprachliche_ausgestaltung.csv', index_col=0)
fgesamteindruck = pd.read_csv('hyperparam-tuning-results/quality-fusion-gesamteindruck.csv', index_col=0)

In [39]:
len(ftextfunktion), len(finhaltliche_ausgestaltung), len(ftextstruktur), len(fsprachliche_ausgestaltung), len(fgesamteindruck)

(5, 5, 5, 5, 5)

In [40]:
ftextfunktion[['test_qwk']].mean(), finhaltliche_ausgestaltung[['test_qwk']].mean(), ftextstruktur[['test_qwk']].mean(), fsprachliche_ausgestaltung[['test_qwk']].mean(), fgesamteindruck[['test_qwk']].mean()

(test_qwk    0.573783
 dtype: float64,
 test_qwk    0.453936
 dtype: float64,
 test_qwk    0.546286
 dtype: float64,
 test_qwk    0.616638
 dtype: float64,
 test_qwk    0.686388
 dtype: float64)

In [41]:
# standard deviation
ftextfunktion[['test_qwk']].std(), finhaltliche_ausgestaltung[['test_qwk']].std(), ftextstruktur[['test_qwk']].std(), fsprachliche_ausgestaltung[['test_qwk']].std(), fgesamteindruck[['test_qwk']].std()

(test_qwk    0.038783
 dtype: float64,
 test_qwk    0.142215
 dtype: float64,
 test_qwk    0.01277
 dtype: float64,
 test_qwk    0.056503
 dtype: float64,
 test_qwk    0.031456
 dtype: float64)

### select representative fold

In [42]:
abs(fgesamteindruck['test_qwk'] - fgesamteindruck['test_qwk'].mean())

0    0.013881
1    0.043547
2    0.013984
3    0.040807
4    0.002843
Name: test_qwk, dtype: float64

In [43]:
abs(fgesamteindruck['test_qwk'] - fgesamteindruck['test_qwk'].mean()).min()

0.0028429196943506874

--> fold = 8 (row 4) is most respresentative

## fusion-settings

### macro

In [44]:
fma_textfunktion = pd.read_csv('hyperparam-tuning-results/quality-fusion-textfunktion-macro.csv', index_col=0)
fma_inhaltliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-inhaltliche_ausgestaltung-macro.csv', index_col=0)
fma_textstruktur = pd.read_csv('hyperparam-tuning-results/quality-fusion-textstruktur-macro.csv', index_col=0)
fma_sprachliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-sprachliche_ausgestaltung-macro.csv', index_col=0)
fma_gesamteindruck = pd.read_csv('hyperparam-tuning-results/quality-fusion-gesamteindruck-macro.csv', index_col=0)

In [45]:
len(fma_textfunktion), len(fma_inhaltliche_ausgestaltung), len(fma_textstruktur), len(fma_gesamteindruck), len(fma_sprachliche_ausgestaltung), 

(5, 5, 5, 5, 5)

In [46]:
fma_textfunktion[['test_qwk']].mean(), fma_inhaltliche_ausgestaltung[['test_qwk']].mean(), fma_textstruktur[['test_qwk']].mean(), fma_sprachliche_ausgestaltung[['test_qwk']].mean(), fma_gesamteindruck[['test_qwk']].mean()

(test_qwk    0.58051
 dtype: float64,
 test_qwk    0.394842
 dtype: float64,
 test_qwk    0.538289
 dtype: float64,
 test_qwk    0.605179
 dtype: float64,
 test_qwk    0.630901
 dtype: float64)

In [47]:
fma_textfunktion[['test_qwk']].std(), fma_inhaltliche_ausgestaltung[['test_qwk']].std(), fma_textstruktur[['test_qwk']].std(), fma_sprachliche_ausgestaltung[['test_qwk']].std(), fma_gesamteindruck[['test_qwk']].std()

(test_qwk    0.027829
 dtype: float64,
 test_qwk    0.064319
 dtype: float64,
 test_qwk    0.030341
 dtype: float64,
 test_qwk    0.048638
 dtype: float64,
 test_qwk    0.040651
 dtype: float64)

### micro

In [48]:
fmi_textfunktion = pd.read_csv('hyperparam-tuning-results/quality-fusion-textfunktion-micro.csv', index_col=0)
fmi_inhaltliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-inhaltliche_ausgestaltung-micro.csv', index_col=0)
fmi_textstruktur = pd.read_csv('hyperparam-tuning-results/quality-fusion-textstruktur-micro.csv', index_col=0)
fmi_sprachliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-sprachliche_ausgestaltung-micro.csv', index_col=0)
fmi_gesamteindruck = pd.read_csv('hyperparam-tuning-results/quality-fusion-gesamteindruck-micro.csv', index_col=0)

In [49]:
len(fmi_textfunktion), len(fmi_inhaltliche_ausgestaltung), len(fmi_textstruktur), len(fmi_sprachliche_ausgestaltung), len(fmi_gesamteindruck)

(5, 5, 5, 5, 5)

In [50]:
fmi_textfunktion[['test_qwk']].mean(), fmi_inhaltliche_ausgestaltung[['test_qwk']].mean(), fmi_textstruktur[['test_qwk']].mean(), fmi_sprachliche_ausgestaltung[['test_qwk']].mean(), fmi_gesamteindruck[['test_qwk']].mean()

(test_qwk    0.565351
 dtype: float64,
 test_qwk    0.43931
 dtype: float64,
 test_qwk    0.531411
 dtype: float64,
 test_qwk    0.581981
 dtype: float64,
 test_qwk    0.655871
 dtype: float64)

In [51]:
fmi_textfunktion[['test_qwk']].std(), fmi_inhaltliche_ausgestaltung[['test_qwk']].std(), fmi_textstruktur[['test_qwk']].std(), fmi_sprachliche_ausgestaltung[['test_qwk']].std(), fmi_gesamteindruck[['test_qwk']].std()

(test_qwk    0.039676
 dtype: float64,
 test_qwk    0.0869
 dtype: float64,
 test_qwk    0.036816
 dtype: float64,
 test_qwk    0.050676
 dtype: float64,
 test_qwk    0.022516
 dtype: float64)

### macro_l1

In [52]:
fma1_textfunktion = pd.read_csv('hyperparam-tuning-results/quality-fusion-textfunktion-macro_l1.csv', index_col=0)
fma1_inhaltliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-inhaltliche_ausgestaltung-macro_l1.csv', index_col=0)
fma1_textstruktur = pd.read_csv('hyperparam-tuning-results/quality-fusion-textstruktur-macro_l1.csv', index_col=0)
fma1_sprachliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-sprachliche_ausgestaltung-macro_l1.csv', index_col=0)
fma1_gesamteindruck = pd.read_csv('hyperparam-tuning-results/quality-fusion-gesamteindruck-macro_l1.csv', index_col=0)

In [53]:
len(fma1_textfunktion), len(fma1_inhaltliche_ausgestaltung), len(fma1_textstruktur), len(fma1_sprachliche_ausgestaltung), len(fma1_gesamteindruck)

(5, 5, 5, 5, 5)

In [54]:
fma1_textfunktion[['test_qwk']].mean(), fma1_inhaltliche_ausgestaltung[['test_qwk']].mean(), fma1_textstruktur[['test_qwk']].mean(), fma1_sprachliche_ausgestaltung[['test_qwk']].mean(), fma1_gesamteindruck[['test_qwk']].mean()

(test_qwk    0.599351
 dtype: float64,
 test_qwk    0.380936
 dtype: float64,
 test_qwk    0.558888
 dtype: float64,
 test_qwk    0.569056
 dtype: float64,
 test_qwk    0.668478
 dtype: float64)

In [55]:
fma1_textfunktion[['test_qwk']].std(), fma1_inhaltliche_ausgestaltung[['test_qwk']].std(), fma1_textstruktur[['test_qwk']].std(), fma1_sprachliche_ausgestaltung[['test_qwk']].std(), fma1_gesamteindruck[['test_qwk']].std()

(test_qwk    0.042927
 dtype: float64,
 test_qwk    0.133572
 dtype: float64,
 test_qwk    0.036433
 dtype: float64,
 test_qwk    0.069338
 dtype: float64,
 test_qwk    0.049335
 dtype: float64)

### macro_l2

In [56]:
fma2_textfunktion = pd.read_csv('hyperparam-tuning-results/quality-fusion-textfunktion-macro_l2.csv', index_col=0)
fma2_inhaltliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-inhaltliche_ausgestaltung-macro_l2.csv', index_col=0)
fma2_textstruktur = pd.read_csv('hyperparam-tuning-results/quality-fusion-textstruktur-macro_l2.csv', index_col=0)
fma2_sprachliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-sprachliche_ausgestaltung-macro_l2.csv', index_col=0)
fma2_gesamteindruck = pd.read_csv('hyperparam-tuning-results/quality-fusion-gesamteindruck-macro_l2.csv', index_col=0)

In [57]:
len(fma2_textfunktion), len(fma2_inhaltliche_ausgestaltung), len(fma2_textstruktur), len(fma2_sprachliche_ausgestaltung), len(fma2_gesamteindruck)

(5, 5, 5, 5, 5)

In [58]:
fma2_textfunktion[['test_qwk']].mean(), fma2_inhaltliche_ausgestaltung[['test_qwk']].mean(), fma2_textstruktur[['test_qwk']].mean(), fma2_sprachliche_ausgestaltung[['test_qwk']].mean(), fma2_gesamteindruck[['test_qwk']].mean()

(test_qwk    0.593291
 dtype: float64,
 test_qwk    0.448431
 dtype: float64,
 test_qwk    0.574785
 dtype: float64,
 test_qwk    0.58053
 dtype: float64,
 test_qwk    0.667674
 dtype: float64)

In [59]:
fma2_textfunktion[['test_qwk']].std(), fma2_inhaltliche_ausgestaltung[['test_qwk']].std(), fma2_textstruktur[['test_qwk']].std(), fma2_sprachliche_ausgestaltung[['test_qwk']].std(), fma2_gesamteindruck[['test_qwk']].std()

(test_qwk    0.029521
 dtype: float64,
 test_qwk    0.105049
 dtype: float64,
 test_qwk    0.018575
 dtype: float64,
 test_qwk    0.054265
 dtype: float64,
 test_qwk    0.035942
 dtype: float64)

### micro_l1

In [60]:
fmi1_textfunktion = pd.read_csv('hyperparam-tuning-results/quality-fusion-textfunktion-micro_l1.csv', index_col=0)
fmi1_inhaltliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-inhaltliche_ausgestaltung-micro_l1.csv', index_col=0)
fmi1_textstruktur = pd.read_csv('hyperparam-tuning-results/quality-fusion-textstruktur-micro_l1.csv', index_col=0)
fmi1_sprachliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-sprachliche_ausgestaltung-micro_l1.csv', index_col=0)
fmi1_gesamteindruck = pd.read_csv('hyperparam-tuning-results/quality-fusion-gesamteindruck-micro_l1.csv', index_col=0)

In [61]:
len(fmi1_textfunktion), len(fmi1_inhaltliche_ausgestaltung), len(fmi1_textstruktur), len(fmi1_sprachliche_ausgestaltung), len(fmi1_gesamteindruck)

(5, 5, 5, 5, 5)

In [62]:
fmi1_textfunktion[['test_qwk']].mean(), fmi1_inhaltliche_ausgestaltung[['test_qwk']].mean(), fmi1_textstruktur[['test_qwk']].mean(), fmi1_sprachliche_ausgestaltung[['test_qwk']].mean(), fmi1_gesamteindruck[['test_qwk']].mean()

(test_qwk    0.600449
 dtype: float64,
 test_qwk    0.437232
 dtype: float64,
 test_qwk    0.543072
 dtype: float64,
 test_qwk    0.585315
 dtype: float64,
 test_qwk    0.663287
 dtype: float64)

In [63]:
fmi1_textfunktion[['test_qwk']].std(), fmi1_inhaltliche_ausgestaltung[['test_qwk']].std(), fmi1_textstruktur[['test_qwk']].std(), fmi1_sprachliche_ausgestaltung[['test_qwk']].std(), fmi1_gesamteindruck[['test_qwk']].std()

(test_qwk    0.025258
 dtype: float64,
 test_qwk    0.136538
 dtype: float64,
 test_qwk    0.044065
 dtype: float64,
 test_qwk    0.053288
 dtype: float64,
 test_qwk    0.046205
 dtype: float64)

### micro_l2

In [64]:
fmi2_textfunktion = pd.read_csv('hyperparam-tuning-results/quality-fusion-textfunktion-micro_l2.csv', index_col=0)
fmi2_inhaltliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-inhaltliche_ausgestaltung-micro_l2.csv', index_col=0)
fmi2_textstruktur = pd.read_csv('hyperparam-tuning-results/quality-fusion-textstruktur-micro_l2.csv', index_col=0)
fmi2_sprachliche_ausgestaltung = pd.read_csv('hyperparam-tuning-results/quality-fusion-sprachliche_ausgestaltung-micro_l2.csv', index_col=0)
fmi2_gesamteindruck = pd.read_csv('hyperparam-tuning-results/quality-fusion-gesamteindruck-micro_l2.csv', index_col=0)

In [65]:
len(fmi2_textfunktion), len(fmi2_inhaltliche_ausgestaltung), len(fmi2_textstruktur), len(fmi2_sprachliche_ausgestaltung), len(fmi2_gesamteindruck)

(5, 5, 5, 5, 5)

In [66]:
fmi2_textfunktion[['test_qwk']].mean(), fmi2_inhaltliche_ausgestaltung[['test_qwk']].mean(), fmi2_textstruktur[['test_qwk']].mean(), fmi2_sprachliche_ausgestaltung[['test_qwk']].mean(), fmi2_gesamteindruck[['test_qwk']].mean()

(test_qwk    0.543686
 dtype: float64,
 test_qwk    0.419792
 dtype: float64,
 test_qwk    0.535106
 dtype: float64,
 test_qwk    0.583367
 dtype: float64,
 test_qwk    0.64455
 dtype: float64)

In [67]:
fmi2_textfunktion[['test_qwk']].std(), fmi2_inhaltliche_ausgestaltung[['test_qwk']].std(), fmi2_textstruktur[['test_qwk']].std(), fmi2_sprachliche_ausgestaltung[['test_qwk']].std(), fmi2_gesamteindruck[['test_qwk']].std()

(test_qwk    0.028197
 dtype: float64,
 test_qwk    0.117536
 dtype: float64,
 test_qwk    0.040793
 dtype: float64,
 test_qwk    0.064147
 dtype: float64,
 test_qwk    0.02339
 dtype: float64)

## significance

In [68]:
from scipy.stats import wilcoxon

better than mDeBERTaV3-adapter

In [69]:
for df in [ftextfunktion, fma_textfunktion, fmi_textfunktion, fma1_textfunktion, fma2_textfunktion, fmi1_textfunktion, fmi2_textfunktion]:
    if wilcoxon(list(atextfunktion['test_qwk']), list(df['test_qwk']), alternative='less').pvalue < 0.05:
        print(df['test_qwk'].mean(), wilcoxon(list(atextfunktion['test_qwk']), list(df['test_qwk']), alternative='less').pvalue)

0.6004485291786701 0.03125


In [70]:
for df in [finhaltliche_ausgestaltung, fma_inhaltliche_ausgestaltung, fmi_inhaltliche_ausgestaltung, fma1_inhaltliche_ausgestaltung, fma2_inhaltliche_ausgestaltung, fmi1_inhaltliche_ausgestaltung, fmi2_inhaltliche_ausgestaltung]:
    if wilcoxon(list(ainhaltliche_ausgestaltung['test_qwk']), list(df['test_qwk']), alternative='less').pvalue < 0.05:
        print(df['test_qwk'].mean(), wilcoxon(list(ainhaltliche_ausgestaltung['test_qwk']), list(df['test_qwk']), alternative='less').pvalue)

In [71]:
for df in [ftextstruktur, fma_textstruktur, fmi_textstruktur, fma1_textstruktur, fma2_textstruktur, fmi1_textstruktur, fmi2_textstruktur]:
    if wilcoxon(list(atextstruktur['test_qwk']), list(df['test_qwk']), alternative='less').pvalue < 0.05:
        print(df['test_qwk'].mean(), wilcoxon(list(atextstruktur['test_qwk']), list(df['test_qwk']), alternative='less').pvalue)

In [72]:
for df in [fsprachliche_ausgestaltung, fma_sprachliche_ausgestaltung, fmi_sprachliche_ausgestaltung, fma1_sprachliche_ausgestaltung, fma2_sprachliche_ausgestaltung, fmi1_sprachliche_ausgestaltung, fmi2_sprachliche_ausgestaltung]:
    if wilcoxon(list(asprachliche_ausgestaltung['test_qwk']), list(df['test_qwk']), alternative='less').pvalue < 0.05:
        print(df['test_qwk'].mean(), wilcoxon(list(asprachliche_ausgestaltung['test_qwk']), list(df['test_qwk']), alternative='less').pvalue)

In [73]:
for df in [fgesamteindruck, fma_gesamteindruck, fmi_gesamteindruck, fma1_gesamteindruck, fma2_gesamteindruck, fmi1_gesamteindruck, fmi2_gesamteindruck]:
    if wilcoxon(list(gesamteindruck['test_qwk']), list(df['test_qwk']), alternative='less').pvalue < 0.05:
        print(df['test_qwk'].mean(), wilcoxon(list(gesamteindruck['test_qwk']), list(df['test_qwk']), alternative='less').pvalue)

0.6863878495364185 0.03125


p-value < 0.05 --> we reject the null hypothesis at a confidence level of 0.03125, concluding that there is a difference in height between the groups. 